spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestDataFrameWrites.java

```diff
@@ -186,21 +185,24 @@ private void writeAndValidateWithLocations(Table table, File location, File expe
     while (expectedIter.hasNext() && actualIter.hasNext()) {
       assertEqualsSafe(tableSchema.asStruct(), expectedIter.next(), actualIter.next());
     }
-    Assert.assertEquals(
-        "Both iterators should be exhausted", expectedIter.hasNext(), actualIter.hasNext());
+    assertThat(actualIter.hasNext())
+        .as("Both iterators should be exhausted")
+        .isEqualTo(expectedIter.hasNext());
 
     table
         .currentSnapshot()
         .addedDataFiles(table.io())
         .forEach(
             dataFile ->
-                Assert.assertTrue(
-                    String.format(
-                        "File should have the parent directory %s, but has: %s.",
-                        expectedDataDir.getAbsolutePath(), dataFile.path()),
-                    URI.create(dataFile.path().toString())
-                        .getPath()
-                        .startsWith(expectedDataDir.getAbsolutePath())));
+                assertThat(
+                        URI.create(dataFile.path().toString())
+                            .getPath()
+                            .startsWith(expectedDataDir.getAbsolutePath()))
```

Comment:
this can be written in a more fluent way (rather than using `isTrue()` at the end)

Reply:
true, fixed!